[ci] Trigger website deploy on new release-data build

captn3m0 opened the pull request:
This triggers a new build, but only on commits to the main branch.

See https://github.com/endoflife-date/endoflife.date/pull/2081 for why

marcwrobel commented:
@captn3m0 , just got a failure after merging #50 : https://github.com/endoflife-date/release-data/actions/runs/3738421224/jobs/6344490740.

captn3m0 commented:
Fixed here: https://github.com/endoflife-date/release-data/commit/e485d315f0d55443e46b00950aff11587778f33e

I've also updated it so that only changes to the JSON files triggers a deploy now: https://github.com/endoflife-date/release-data/commit/44e95ac39a3cfe51bfbfd3a0260504dbd30ec3da

marcwrobel commented:
OK, cool !
